crates/shared/src/price_estimation/instrumented.rs

```diff
@@ -58,6 +75,32 @@ impl PriceEstimating for InstrumentedPriceEstimator {
     }
 }
 
+impl<T: NativePriceEstimating> NativePriceEstimating for InstrumentedPriceEstimator<T> {
+    fn estimate_native_price(&self, token: H160) -> BoxFuture<'_, NativePriceEstimateResult> {
+        async move {
+            let start = Instant::now();
+            let estimate = self.inner.estimate_native_price(token).await;
+            self.metrics
+                .native_price_estimation_times
+                .with_label_values(&[self.name.as_str()])
+                .observe(start.elapsed().as_secs_f64());
+            let result = self.estimate_result(estimate.as_ref());
+            self.metrics
+                .native_price_estimates
+                .with_label_values(&[self.name.as_str(), result])
+                .inc();
```

Comment:
Do we really need 2 metrics? Can we use the `native_price_estimation_times` only with 2 labels and use the `_count` metric of it instead of declaring one more with increment?